homeassistant/components/esphome/config_flow.py

```diff
@@ -53,6 +53,23 @@ def __init__(self):
             errors=errors
         )
 
+    async def async_step_discovery(self, user_input: ConfigType):
+        """Handle discovery."""
+        # mDNS hostname has additional '.' at end
+        hostname = user_input['hostname'].replace('.local.', '.local')
+        hosts = (hostname, user_input['host'])
```

Comment:
Why check for both?

Reply:
If a user has manually configured an ESPHome node with a static IP, they shouldn't get a "ESPHome" discovered message. For example this is good to improve WiFi connection times (DHCP is _slow_)

Similarly, a user can configure an ESPHome node via `livingroom.local`, if that already exist we also shouldn't create a discovery message.